New Update - Implementation for SIRS Model

kle6951 opened the pull request:
I have done the implementation for the solver for the SIRS model #17. Please note that it is not runnable right now because of a bug. I am looking for the bug coming from. 

**_There are a few things that need to be considered:_** 
1. There is a lack of data and assigned values for each parameter for SIRS, therefore, I need verification and resources for the data for this model
2. I understand that there will be more works to have all the models to work completely in general. Right now, there is a duplicate code (ie: plotTheme) in every model that needs us to take note of having another plot file that works for all models.

Please review the code and let me know what I can do to improve the code and correct the code if needed!

bryce-carson commented:
> Please review the code and let me know what I can do to improve the code and correct the code if needed!

Thanks! I am beginning the review.

bryce-carson commented on R/SIRS_solve.R:
```diff
@@ -0,0 +1,86 @@
+                  x = "Time") +
+    ggplot2::scale_x_continuous(expand = c(0, 0)) +
+    ggplot2::scale_y_continuous(expand = c(0, 0)) +
+    ggplot2::geom_line(aes(y = S, color = "Blue"), linewidth = 1.5) +
+    ggplot2::geom_line(aes(y = I, color = "Red"), linewidth = 1.5) +
+    ggplot2::geom_line(aes(y = R, color = "Green"), linewidth = 1.5) +
```
Please include `ggplot2::` when calling `aes()` here.

bryce-carson commented on R/SIRS_solve.R:
```diff
@@ -0,0 +1,86 @@
+    ggplot2::geom_line(aes(y = I, color = "Blue"), linewidth = 1.5) +
+    plotTheme +
+    ggplot2::ggtitle("SI Phase Plane") + 
+    ggplot2::theme(plot.title = element_text(size = 22, face = "bold")) +
```
Please include `ggplot2::` when calling `element_text()` here.

bryce-carson commented on server.R:
Thanks for including the SIRS validation section here, but the changes will not be needed. To make work progress smoothly, please modify this PR _by [ammending this commit](https://github.com/ashokkrish/episim/pull/32/commits/6b9ac0d27973dcc3d818339f9e815dba4d9a3985) and dropping the changes to server.R_. Do not revert the commit or delete the file, you must ammend the commit and drop the staged changes to the server.R file.

Thanks!

kle6951 replied:
How can I ammend this commit and drop the changes to server.R?

bryce-carson replied:
I'm sorry, I misguided you. You don't need to do this anymore, I will rebase my changes onto the copy of the server you push in this PR.

Thanks for asking for help, regardless, @kle6951.

kle6951 replied:
Great! Let me know if you need me to do anything.

bryce-carson reviewed (changes requested):
Thanks, @kle6951! Please request another review of the PR when you've addressed all the requested changes.

kle6951 commented:
I fixed the bugs in my code for the SIRS solver and corrected the syntax by adding the ggplot2. One thing I have not done is amend the commit and drop the change to the server.R . Please let me know how should I do it.

bryce-carson commented on R/SIRS_solve.R:
```diff
@@ -18,17 +18,25 @@ equationsSIRS <- function(time, variables, parameters)
-  deSolve::lsoda(variables, seq(0, timesteps, by = 1),
-                 equationsSIR, parameters) |> as.data.frame()
-}
+solveSIRS <-
```
Good catch here, I didn't notice that it was being assigned to a function previously.

bryce-carson commented:
> **_There are a few things that need to be considered:_**
> 
>     1. There is a lack of data and assigned values for each parameter for SIRS, therefore, I need verification and resources for the data for this model

I'm not sure what you meant, but the model will run with the values passed to the function implementing the model. It has values in that way.

>     2. I understand that there will be more works to have all the models to work completely in general. Right now, there is a duplicate code (ie: plotTheme) in every model that needs us to take note of having another plot file that works for all models.

Yes, we will be integrating the models and reducing code duplication soon. We need to meet and discuss these things in person because the coordination of some things is difficult, and this is my first time leading a project, so I don't want to make more mistakes. :)
